flink-connectors/flink-connector-kafka/src/test/java/org/apache/flink/streaming/connectors/kafka/table/KafkaDynamicTableFactoryTest.java

```diff
@@ -679,106 +729,165 @@ private void verifyEncoderSubject(
 
     @Test
     public void testSourceTableWithTopicAndTopicPattern() {
-        thrown.expect(ValidationException.class);
-        thrown.expect(
-                containsCause(
-                        new ValidationException(
-                                "Option 'topic' and 'topic-pattern' shouldn't be set together.")));
-
-        final Map<String, String> modifiedOptions =
-                getModifiedOptions(
-                        getBasicSourceOptions(),
-                        options -> {
-                            options.put("topic", TOPICS);
-                            options.put("topic-pattern", TOPIC_REGEX);
-                        });
-
-        createTableSource(SCHEMA, modifiedOptions);
+        assertThatThrownBy(
+                        () -> {
+                            final Map<String, String> modifiedOptions =
+                                    getModifiedOptions(
+                                            getBasicSourceOptions(),
+                                            options -> {
+                                                options.put("topic", TOPICS);
+                                                options.put("topic-pattern", TOPIC_REGEX);
+                                            });
+
+                            createTableSource(SCHEMA, modifiedOptions);
+                        })
+                .isInstanceOf(ValidationException.class)
+                .satisfies(
+                        throwable ->
+                                assertThatChainOfCauses(throwable)
+                                        .anySatisfy(
+                                                cause ->
+                                                        assertThat(cause)
+                                                                .isInstanceOf(
+                                                                        ValidationException.class)
+                                                                .hasMessage(
+                                                                        "Option 'topic' and 'topic-pattern' shouldn't be set together.")));
```

Comment:
For the nested exceptions, you can use Flink's util `ExceptionUtils#findThrowable` or `FlinkAssertions.anyCauseMatches()`

Reply:
`FlinkAssertions.anyCauseMatches()` uses `hasMessageContaining` to check the error message. We need to check with `hasMessage` here.
I will add a new method `containsCause` to `FlinkAssertions` to fix it.